ruff_cli/src/printer.rs

```diff
@@ -336,6 +343,55 @@ impl<'a> Printer<'a> {
         Ok(())
     }
 
+    pub fn write_statistics(&self, diagnostics: &Diagnostics) -> Result<()> {
+        let mut violations = diagnostics
+            .messages
+            .iter()
+            .map(|message| message.kind.rule())
+            .collect::<Vec<_>>();
+        violations.sort();
+        violations.dedup();
+        let statistics = violations.iter().map(|rule| ExpandedStatistics {
+            code: rule.code(),
+            count: diagnostics
+                .messages
+                .iter()
+                .filter(|message| message.kind.rule() == *rule)
+                .count(),
+            message: diagnostics
+                .messages
+                .iter()
+                .find(|message| message.kind.rule() == *rule)
+                .map(|message| message.kind.body())
+                .unwrap(),
+        });
+
+        let mut stdout = BufWriter::new(io::stdout().lock());
+        match self.format {
+            SerializationFormat::Json => {
+                writeln!(
+                    stdout,
+                    "{}",
+                    serde_json::to_string_pretty(&statistics.collect::<Vec<_>>())?
+                )?;
+            }
+            SerializationFormat::Text => {
+                for msg in statistics {
+                    // same format as flake8 has
+                    writeln!(stdout, "{:<5} {} {}", msg.count, msg.code, msg.message)?;
+                }
+                return Ok(());
+            }
+            _ => {
+                anyhow::bail!("Unsuported format for --statistics!")
```

Comment:
It probably makes sense to mention which formats are supported in the error message.

Reply:
done